mico-core/src/main/java/io/github/ust/mico/core/model/MicoLabel.java

```diff
@@ -19,31 +19,68 @@
 
 package io.github.ust.mico.core.model;
 
-import java.util.Map;
-
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
-
+import io.github.ust.mico.core.configuration.extension.CustomOpenApiExtentionsPlugin;
+import io.github.ust.mico.core.util.Patterns;
+import io.swagger.annotations.ApiModelProperty;
+import io.swagger.annotations.Extension;
+import io.swagger.annotations.ExtensionProperty;
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.NoArgsConstructor;
 import lombok.experimental.Accessors;
 
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+import java.util.Map;
+
 /**
  * Represents a simple key-value pair label.
  * Necessary since Neo4j does not allow to persist
  * {@link Map} implementations.
  *
- * @param <K> the type of the key.
- * @param <V> the type of the value.
  */
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
 @Accessors(chain = true)
 @JsonIgnoreProperties(ignoreUnknown = true)
-public class MicoLabel<K, V> {
-    
-    private K key;
-    private V value;
+public class MicoLabel {
+
+    /**
+     * Key of the label.
+     */
+    @ApiModelProperty(required = true, extensions = {@Extension(
+        name = CustomOpenApiExtentionsPlugin.X_MICO_CUSTOM_EXTENSION,
+        properties = {
+            @ExtensionProperty(name = "title", value = "Key"),
+            @ExtensionProperty(name = "pattern", value = Patterns.KUBERNETES_LABEL_KEY_REGEX),
+            @ExtensionProperty(name = "minLength", value = "1"),
+            @ExtensionProperty(name = "maxLength", value = "253"),
+            @ExtensionProperty(name = "x-order", value = "10"),
+            @ExtensionProperty(name = "description", value = "Key of the label.")
+        }
+    )})
+    @Size(min = 1, max = 253, message = "must have a length between 1 and 63 without prefix or a length between 1 and 253 with prefix")
+    @Pattern(regexp = Patterns.KUBERNETES_LABEL_KEY_REGEX, message = Patterns.KUBERNETES_LABEL_KEY_MESSAGE)
+    private String key;
+
+    /**
+     * Value of the label.
+     */
+    @ApiModelProperty(extensions = {@Extension(
+        name = CustomOpenApiExtentionsPlugin.X_MICO_CUSTOM_EXTENSION,
+        properties = {
+            @ExtensionProperty(name = "title", value = "Value"),
+            @ExtensionProperty(name = "pattern", value = Patterns.KUBERNETES_LABEL_VALUE_REGEX),
+            @ExtensionProperty(name = "minLength", value = "0"),
+            @ExtensionProperty(name = "maxLength", value = "63"),
+            @ExtensionProperty(name = "x-order", value = "20"),
+            @ExtensionProperty(name = "description", value = "Value of the label.")
+        }
+    )})
+    @Size(max = 63, message = "must be 63 characters or less")
```

Comment:
Where does this restriction come from and is this relevant for the ui?

Reply:
Currently not relevant for the UI.